### PR TITLE
Add mm-test of mariadb ssl connection

### PR DESCRIPTION
**Pull request by mitiao**
See: https://progress.opensuse.org/issues/23348

Verified runs:
sle-fips:
http://10.67.17.71/tests/2
http://10.67.17.71/tests/3
tumbleweed:
http://10.67.17.71/tests/21
http://10.67.17.71/tests/22

Needle:
https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/458

### CODE REVIEW

File: tests/console/mariadb.pm [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+use strict;
+use testapi;
+use lockapi;
+use utils;
+use mm_tests;
+
+sub run {
+    select_console 'root-console';
+    configure_static_network('10.0.2.11/24');
+    zypper_call('in mariadb');
```
Comment (kravciak) [inline tests/console/mariadb.pm]: Maybe "in mariadb-client" would be enough

File: tests/console/mariadb.pm [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    mutex_lock('mariadb');
+    mutex_unlock('mariadb');
+
+    type_string_slow "mysql --ssl --host=10.0.2.1 --user=root --password\n";
+    assert_screen 'mariadb-monitor-password';
+    type_string_slow "suse\n";
+    # Give more time to authenticate and open the db server
+    assert_screen 'mariadb-monitor-opened', 60;
+    type_string_slow "quit\n";
```
Comment (kravciak) [inline tests/console/mariadb.pm]: No need to "quit" here if there is no follow-up check - but it's up to your preference

Comment (okurz) [inline tests/console/mariadb.pm]: I would keep the quit as proper cleanup

**Comment by okurz** [comment]
please test on openSUSE Tumbleweed as well

File: tests/console/mariadb.pm (hunk 1 above)
Comment (mitiao) [inline tests/console/mariadb.pm]: Yeah, should be

**Comment by mitiao** [comment]
This case is default loaded in FIPS mmtest:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/products/sle/main.pm#L1206

For Tumbleweed I will take time to test and check where we should put this and even all cases to product/opensuse/main.pm.

**Comment by mitiao** [comment]
Added entry and tested on openSUSE Tumbleweed. 
And for the entry of non-multi-machine tests(fips test sets on sle), I will test them on Tumbleweed and find proper place to load in next pull request.

**Comment by okurz** [comment]
I see, good.

**Comment by okurz** [comment]
I am not sure if we have any multi machine test on o3 at all for now (including the necessary infrastructure) so that would need to be checked.

**Comment by mitiao** [comment]
Yesterday I tried and did not find any multi machine from the dashboard of o3.
And I would add and run 1 mm test to see how it work on o3, could you grant me the admin permission for it?

**Comment by okurz** [comment]
So far the best practice regarding admin permissions on o3 is that the persons should stay active in [#opensuse-factory](irc://chat.freenode.net/opensuse-factory). If you agree please ask me there. OTOH I don't think it can work when we don't have a tap worker but of course we can just try, e.g. by adding a test suite and jobs to the openSUSE Tumbleweed development group.

**Pull request merged**